src/pages/ProfilePage.tsx

```diff
@@ -21,154 +20,136 @@ import PressableWithoutFocus from '@components/Pressable/PressableWithoutFocus';
-    return details.login ? Str.removeSMSDomain(details.login) : '';
+    return login ? Str.removeSMSDomain(login) : '';
 };
 
-function ProfilePage(props) {
+function ProfilePage({personalDetails = {}, route, session, report}: ProfilePageProps) {
     const styles = useThemeStyles();
-    const accountID = Number(lodashGet(props.route.params, 'accountID', 0));
-    const details = lodashGet(props.personalDetails, accountID, ValidationUtils.isValidAccountRoute(accountID) ? {} : {isloading: false});
+    const {translate, formatPhoneNumber} = useLocalize();
+    const accountID = Number(route.params?.accountID ?? 0);
+    const details: PersonalDetails =
+        personalDetails?.[accountID] ?? (ValidationUtils.isValidAccountRoute(accountID) ? {accountID: 0, avatar: ''} : {isLoading: false, accountID: 0, avatar: ''});
 
     const displayName = PersonalDetailsUtils.getDisplayNameOrDefault(details);
-    const avatar = lodashGet(details, 'avatar', UserUtils.getDefaultAvatar());
-    const fallbackIcon = lodashGet(details, 'fallbackIcon', '');
-    const originalFileName = lodashGet(details, 'originalFileName', '');
-    const login = lodashGet(details, 'login', '');
-    const timezone = lodashGet(details, 'timezone', {});
+    const avatar = details?.avatar ?? UserUtils.getDefaultAvatar();
+    const fallbackIcon = details?.fallbackIcon ?? '';
+    const originalFileName = details?.originalFileName ?? '';
+    const login = details?.login ?? '';
+    const timezone = details?.timezone;
 
     // If we have a reportID param this means that we
     // arrived here via the ParticipantsPage and should be allowed to navigate back to it
-    const shouldShowLocalTime = !ReportUtils.hasAutomatedExpensifyAccountIDs([accountID]) && !_.isEmpty(timezone);
-    let pronouns = lodashGet(details, 'pronouns', '');
-    if (pronouns && pronouns.startsWith(CONST.PRONOUNS.PREFIX)) {
+    const shouldShowLocalTime = !ReportUtils.hasAutomatedExpensifyAccountIDs([accountID]) && !isEmptyObject(timezone);
+    let pronouns = details?.pronouns ?? '';
+    if (pronouns?.startsWith(CONST.PRONOUNS.PREFIX)) {
         const localeKey = pronouns.replace(CONST.PRONOUNS.PREFIX, '');
-        pronouns = props.translate(`pronouns.${localeKey}`);
+        pronouns = translate(`pronouns.${localeKey}` as TranslationPaths);
     }
 
     const isSMSLogin = Str.isSMSLogin(login);
     const phoneNumber = getPhoneNumber(details);
     const phoneOrEmail = isSMSLogin ? getPhoneNumber(details) : login;
 
-    const isCurrentUser = props.session.accountID === accountID;
-    const hasMinimumDetails = !_.isEmpty(details.avatar);
-    const isLoading = lodashGet(details, 'isLoading', false) || _.isEmpty(details);
+    const isCurrentUser = session?.accountID === accountID;
+    const hasMinimumDetails = !isEmptyObject(details.avatar);
+    const isLoading = details?.isLoading ?? false ?? isEmptyObject(details);
```

Comment:
Current version will never reach the last option cause nullish coalescing will use `false` as a value
```suggestion
    const isLoading = !!details?.isLoading || isEmptyObject(details);
```